fluss-common/src/main/java/com/alibaba/fluss/lake/lakestorage/PluginLakeStorageWrapper.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.lake.lakestorage;
+
+import com.alibaba.fluss.config.Configuration;
+import com.alibaba.fluss.exception.TableAlreadyExistException;
+import com.alibaba.fluss.lake.writer.LakeTieringFactory;
+import com.alibaba.fluss.metadata.TableDescriptor;
+import com.alibaba.fluss.metadata.TablePath;
+import com.alibaba.fluss.utils.TemporaryClassLoaderContext;
+import com.alibaba.fluss.utils.WrappingProxy;
+
+/**
+ * A wrapper around {@link LakeStoragePlugin} that ensures the plugin classloader is used for all
+ * {@link LakeCatalog} and {@link LakeStorage} operations.
+ */
+public class PluginLakeStorageWrapper implements LakeStoragePlugin {
```

Comment:
Sorry for missing that. Can we wrap `LakeCatalog` instead of `LakeStoragePlugin`, only `LakeCatalog` will be called in server side. So, only `LakeCatalog` matters. Also, currently, it doesn't wrap the `LakeCatalog#catalogtable` method, which I think may need to set thread current context.